analytic_engine/src/instance/serializer.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    sync::{Arc, Condvar, Mutex},
+    time::Instant,
+};
+
+use common_util::{runtime::Runtime, time::InstantExt};
+use futures::Future;
+use log::error;
+use table_engine::table::{Error as TableError, Result as TableResult, TableId};
+use tokio::sync::oneshot;
+
+use crate::{
+    instance::flush_compaction::{self, BackgroundFlushFailed},
+    table::metrics::Metrics,
+};
+
+#[derive(Default)]
+enum FlushState {
+    #[default]
+    Ok,
+    Flushing,
+    Failed {
+        err_msg: String,
+    },
+}
+
+type ScheduleLock = Arc<(Mutex<FlushState>, Condvar)>;
```

Comment:
How about 
```
struct {
    flush_state: Mutex<FlushState>,
    flush_notifier: Condvar,
}
```

Reply:
Isn't a channel more simple than use a condition variable here?

Condvar is a lower-level API, I think we could avoid it when possible.

Reply:
I have modified it to `tokio::sync::watch::channel`.